.azure-pipelines/templates/platform_pr_pipeline.yml

```diff
@@ -24,4 +25,22 @@ stages:
           script_parameter: |
             python -m mlops.${{ parameters.model_type }}.src.mlops_pipeline \
               --build_environment ${{parameters.exec_environment}} \
-              --wait_for_completion True
+              --wait_for_completion True \
+              --output_file run_id.txt
+      - task: AzureCLI@2
+        displayName: "Cancel AML Pipeline"
+        condition: canceled()
+        inputs:
+          azureSubscription: $(AZURE_RM_SVC_CONNECTION)
+          scriptType: bash
+          scriptLocation: inlineScript
+          inlineScript: |
+            readarray arr <"run_id.txt"
+            run_name=${arr[0]}
+
+            echo "Pipeline was canceled. Attempting to cancel AML Pipeline: ${run_name}..."
+
+            az extension add --name ml
+            az ml job cancel --name $run_name \
```

Comment:
It's usually a good practice to quote shell variable references because if they contain spaces or special characters, it can cause unwanted behavior.

This would be "$run_name"